vignettes/visits_periods.Rmd

```diff
@@ -0,0 +1,273 @@
+    APHASEN = as.integer(APHASEN),
+    across(matches("PH[ES]DT"), ymd)
+  ) %>%
+  select(STUDYID, everything())
+dataset_vignette(
+  phase_ref
+)
+```
+
+The admiral functions expect separate datasets for periods, subperiods, and
+phases. For periods the numeric variable `APERIOD` is expected, for subperiods
+the numeric variables `APERIOD` and `ASPER`, and for phases the numeric variable
+`APHASEN`.
+
+## Creating ADSL Period, Subperiod, or Phase Variables {#periods_adsl}
+
+If a period reference dataset is available, the ADSL variables for periods,
+subperiods, or phases can be created from this dataset by calling
+`derive_vars_period()`.
+
+For example the period reference dataset from the previous section can be used
+to add the phase variables (`PHwSDT`, `PHwEDT`, and `APHASEw`) to ADSL:
+```{r}
+adsl <- tibble(STUDYID = "xyz", USUBJID = c("1", "2"))
+
+adsl <- derive_vars_period(
+  adsl,
+  dataset_ref = phase_ref,
+  new_vars = vars(PHwSDT = PHSDT, PHwEDT = PHEDT, APHASEw = APHASE)
+) %>%
+  select(STUDYID, USUBJID, PH1SDT, PH1EDT, PH2SDT, PH2EDT, APHASE1, APHASE2)
+
+adsl
+```
+
+## Creating BDS and OCCDS Period, Subperiod, or Phase Variables {#periods_bds}
+
+If a period reference dataset is available, BDS and OCCDS variables for periods,
+subperiods, or phases can be created by calling `derive_vars_joined()`.
+
+For example the variables `APHASEN`, `PHSDT`, `PHEDT`, `APHASE` can be derived
+from the period reference dataset defined above.
+
+```{r}
+adae <- tribble(
+  ~USUBJID, ~ASTDT,
+  "1",      "2021-01-01",
+  "1",      "2021-01-05",
+  "1",      "2021-02-05",
+  "1",      "2021-03-05",
+  "1",      "2021-04-05",
+  "2",      "2021-02-15",
+) %>%
+  mutate(ASTDT = ymd(ASTDT))
+
+derive_vars_joined(
+  adae,
+  dataset_add = phase_ref,
+  by_vars = vars(USUBJID),
+  filter_join = PHSDT <= ASTDT & ASTDT <= PHEDT
+)
+```
+
+If no period reference dataset is available but period variables are in ADSL,
+the period reference dataset can be created from ADSL by calling
+`create_period_dataset()`.
+
+For example, a period reference dataset for phases can be created from the ADSL
+dataset created above:
+```{r}
+create_period_dataset(
+  adsl,
+  new_vars = vars(PHSDT = PHwSDT, PHEDT = PHwEDT, APHASE = APHASEw)
+)
+```
+
+# Treatment Variables (`TRTxxP`, `TRTxxA`, `TRTP`, `TRTA`, ...)
+
+In studies with multiple periods the treatment can differ by period. CDISC
+defines variables for planned and actual treatments in ADSL (`TRTxxP`, `TRTxxA`,
+`TRxxPGy`, `TRxxAGy`, ...) and corresponding variables in BDS and OCCDS datasets
+(`TRTP`, `TRTA`, `TRTPGy`, `TRTAGy`, ...). They can be derived in the same way
+(and same step) as the period, subperiod, and phase variables.
+
+## Creating ADSL Treatment Variables {#treatment_adsl}
+
+If the treatment information is included in the period reference dataset, the
+treatment ADSL variables can be created by calling `derive_vars_periods()`:
+
+```{r}
+# Add period variables to ADSL
```

Comment:
i wonder if these examples would be simpler for users to follow if you only included TRTA instead of also TRTP - i'd imagine that will be the most common use case. happy for you to make the call though

Reply:
I agree, `TRTP` removed.